src/ServiceControl/Operations/ConnectedApplicationsEnricher.cs

```diff
@@ -0,0 +1,27 @@
+﻿namespace ServiceControl.Operations
+{
+    using System.Collections.Generic;
+    using System.Linq;
+    using ServiceControl.Persistence;
+
+    class ConnectedApplicationsEnricher(IConnectedApplicationsDataStore connectedApplicationsDataStore) : IEnrichImportedErrorMessages
+    {
+        static List<string> MassTransitHeaders =
+        [
+            "MassTransitHeader1",
+            "MassTransitHeader2",
+            "MassTransitHeader3"
+        ];
+
+        public void Enrich(ErrorEnricherContext context)
+        {
+            if (!alreadyAddedMassTransit && context.Headers.Any(incomingHeader => MassTransitHeaders.Contains(incomingHeader.Key)))
```

Comment:
This is confusing. It tries to be both generic (`ConnectedApplicationsEnricher`) and specific (`alreadyAddedMassTransit`)

I think we should go: 

1. Be specific and have an enricher per "application/component/feature" and check for the existence of any `MT-%` or probe a subset of headers like the current implementation and call this `MatchMassTransitEnricher`
2. A bit more generic look for a specific header like `Feature` and capture all values but that requires the sender to add a header that contains the product identifier
3. Make it even more generic and iterate over a bunch of registered `FeatureProbe` and for each match store each corresponding "identifier" so that the task of the enricher is to take care of storing and caching any matches and have very simple "probe" implementations

Reply:
I envisaged this extending to other applications. e.g.

```cs
        static string MassTransitHeaderPrefix = "MT-";
        static string MicrosoftEventingFrameworkHeaderPrefix = "MS-";

        public void Enrich(ErrorEnricherContext context)
        {
            if (!alreadyAddedMassTransit && context.Headers.Any(incomingHeader => incomingHeader.Key.StartsWith(MassTransitHeaderPrefix, System.StringComparison.InvariantCultureIgnoreCase)))
            {
                alreadyAddedMassTransit = true;
                _ = connectedApplicationsDataStore.Add("MassTransitConnector");
            } else if (!alreadyAddedMicrosoftEventingFramework && context.Headers.Any(incomingHeader => incomingHeader.Key.StartsWith(MicrosoftEventingFrameworkHeaderPrefix , System.StringComparison.InvariantCultureIgnoreCase)))
            {
                alreadyAddedMicrosoftEventingFramework = true;
                _ = connectedApplicationsDataStore.Add("MicrosoftEventingFramework");
            }
        }

        static bool alreadyAddedMassTransit = false;
        static bool alreadyAddedMicrosoftEventingFramework = false;

```